src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Core/IdeaCollection/Workbench/Workbench.ts

```diff
@@ -226,6 +236,90 @@ export var registerRoutesFactory = (
+                    };
+                }])
+            .defaultVersionable(RIDocument, RIDocumentVersion, "edit", ideaCollection.content_type, context, {
+                space: "content",
+                movingColumns: "is-show-show-hide"
+            })
+            .specificVersionable(RIDocument, RIDocumentVersion, "edit", ideaCollection.content_type, context, [
+                "adhHttp", (adhHttp : AdhHttp.Service) => (item : ResourcesBase.IResource, version : ResourcesBase.IResource) => {
+                    return adhHttp.options(item.path).then((options : AdhHttp.IOptions) => {
+                        if (!options.POST) {
+                            throw 401;
+                        } else {
+                            return {
+                                documentUrl: version.path
+                            };
+                        }
+                    });
+                }])
+        .defaultVersionable(RIParagraph, RIParagraphVersion, "comments", ideaCollection.content_type, context, {
+            space: "content",
+            movingColumns: "is-collapse-show-show"
+        })
+        .specificVersionable(RIParagraph, RIParagraphVersion, "comments", ideaCollection.content_type, context, [
+            () => (item : ResourcesBase.IResource, version : ResourcesBase.IResource) => {
+                var documentUrl = _.last(_.sortBy(SIParagraph.get(version).documents));
+                return {
+                    commentableUrl: version.path,
+                    commentCloseUrl: documentUrl,
+                    documentUrl: documentUrl
+                };
+            }])
+        .defaultVersionable(RIComment, RICommentVersion, "", ideaCollection.content_type, context, {
+            space: "content",
+            movingColumns: "is-collapse-show-show"
+        })
+        .specificVersionable(RIComment, RICommentVersion, "", ideaCollection.content_type, context, ["adhHttp", "$q", (
+            adhHttp : AdhHttp.Service,
+            $q : angular.IQService
+        ) => {
+            var getCommentableUrl = (resource) : angular.IPromise<any> => {
+                if (resource.content_type !== RICommentVersion.content_type) {
+                    return $q.when(resource);
+                } else {
+                    var url = SIComment.get(resource).refers_to;
+                    return adhHttp.get(url).then(getCommentableUrl);
+                }
+            };
+
+            return (item : ResourcesBase.IResource, version : ResourcesBase.IResource) => {
+                return getCommentableUrl(version).then((commentable) => {
+                    var documentUrl = _.last(_.sortBy(SIParagraph.get(commentable).documents));
+                    return {
+                        commentableUrl: commentable.path,
+                        commentCloseUrl: documentUrl,
+                        documentUrl: documentUrl
+                    };
+                });
+            };
+        }]);
+    }
```

Comment:
I think it would be clearer to put the proposal/document specific routes into separate functions, maybe even into the respective modules.